### PR TITLE
Fix: ESO template crash when Kubernetes secret data is nil 

### LINKED ISSUE #3294: ESO pod crashes if the secret referenced in the PushSecret has no data
**Describe the bug**
If you create a PushSecret that references a secret without a data key, the ESO pod crashes with the error message 

```
{"level":"info","ts":1711105214.26202,"msg":"Observed a panic in reconciler: assignment to entry in nil map","controller":"pushsecret","controllerGroup":"external-secrets.io","controllerKind":"PushSecret","PushSecret":{"name":"my-secret","namespace":"test"},"namespace":"test","name":"my-secret","reconcileID":"566de87-9b33-403c-9006-adbece35940a"}
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map
```

**To Reproduce**
1. Install ESO with Version v0.9.13
2. Create Secret without data field

```
apiVersion: v1
kind: Secret
metadata:
  name: my-secret
  namespace: test
```

3. Create PushSecret 

```
apiVersion: external-secrets.io/v1alpha1
kind: PushSecret
metadata:
  name: my-secret
  namespace: test
spec:
  deletionPolicy: Delete
  refreshInterval: 10s
  secretStoreRefs:
    - name: build
      kind: SecretStore
  selector:
    secret:
      name: my-secret
  template:
    data:
      test: "{{ .test }}"
  data:
    - match:  
        secretKey: test
        remoteRef:
          remoteKey: test
          property: test
```

**Expected behavior**
My expectation is that the pod will not crash when a secret referenced by a PushSecret has no data field.

### PR DESCRIPTION
**Pull request by ma-ble**
## Problem Statement

If you create a PushSecret that references a secret without a data key and uses it in the template, the eso pod crashes.

## Related Issue

Fixes #3294

## Proposed Changes

Add nil pointer check before assigning a value.

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

File: pkg/template/v2/template.go [hunk 1]
```diff
@@ -76,7 +76,9 @@ func applyToTarget(k, val string, target esapi.TemplateTarget, secret *corev1.Se
 	case esapi.TemplateTargetLabels:
 		secret.Labels[k] = val
 	case esapi.TemplateTargetData:
-		secret.Data[k] = []byte(val)
+		if secret.Data != nil {
```
[the remark this answers is not in the document]
Comment (ma-ble) [reply pkg/template/v2/template.go:79]: I'm not sure if I understand your point exactly. "secret.Data" corresponds to the optional data field from the Kubernetes Secret, which is referenced in the PushSecret. If the secret doesn't have a data field, we can't add anything from the template engine.

[the remark this answers is not in the document]
Comment (ma-ble) [reply pkg/template/v2/template.go]: I'm not entirely sure about the use case, but similar to externalsecret, you can place the check in front of the template engine and then create an empty map. I adjusted the PR accordingly.

**Review by moolen — COMMENTED** [review]
Hey, i re-read the referenced issue #3294. Looking at stack traces and context provided there i suggest to change the function which modifies the Data/Labels/Annotations here and add a guard which creates a map if needed:

https://github.com/external-secrets/external-secrets/blob/c3bf442941ffa91d230f9dafec0bbc2372a57906/pkg/template/v2/template.go#L72-L82

Not only could the `.Data` be nil, also Annotations or Labels could be nil.

**Review by moolen — APPROVED** [review]
LGTM! Thanks for the tests as well :bow:

**Pull request merged**